src/utils/response/response.ts

```diff
@@ -0,0 +1,28 @@
+import { PagingParams } from '../params/pagingParams';
+import { RequestParams } from '../params/requestParams';
+import { RequestPromise, toApiResponse } from './responseHandlers';
+
+export type Response<T> = {
+  success: boolean;
+  status?: number;
+  results: T[];
+  startTime: string;
+  requestParams: RequestParams<T>;
+  prevPage?: PagingParams<T>;
+  currentPage?: PagingParams<T>;
+  nextPage?: PagingParams<T>;
+};
```

Comment:
Not sure regarding the plans with `ApiResponse`, one option to not duplicate those response fields would be defining `Response<T>` like this:

```suggestion
export type Response<T> = ApiResponseOnSuccess<T[]> & {
  requestParams: RequestParams<T>;
  prevPage?: PagingParams<T>;
  currentPage?: PagingParams<T>;
  nextPage?: PagingParams<T>;
};
```

Reply:
I chose to have an own response type because ApiResponse uses T[] which makes no sense for this use case. 
and the naming conversion of ApiResponse doesn't look like we want to return this to the user.  member names start with upper case.